pkg/medusa/reconcile.go

```diff
@@ -518,6 +524,52 @@ func StandaloneMedusaService(dcConfig *cassandra.DatacenterConfig, medusaSpec *a
 	return medusaService
 }
 
+func PurgeCronJob(dcConfig *cassandra.DatacenterConfig, medusaSpec *api.MedusaClusterTemplate, clusterName, namespace string, logger logr.Logger) *batchv1.CronJob {
+	purgeCronJob := &batchv1.CronJob{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      MedusaPurgeCronJobName(cassdcapi.CleanupForKubernetes(clusterName), dcConfig.SanitizedName()),
+			Namespace: namespace,
+		},
+		Spec: batchv1.CronJobSpec{
+			Schedule:                   "0 0 * * *",
+			Suspend:                    pointer.Bool(false),
+			SuccessfulJobsHistoryLimit: pointer.Int32(3),
+			FailedJobsHistoryLimit:     pointer.Int32(1),
+			JobTemplate: batchv1.JobTemplateSpec{
+				Spec: batchv1.JobSpec{
+					Template: corev1.PodTemplateSpec{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "k8ssandra-purge-backups",
+						},
+						Spec: corev1.PodSpec{
+							RestartPolicy:                 corev1.RestartPolicyOnFailure,
+							TerminationGracePeriodSeconds: pointer.Int64(30),
+							DNSPolicy:                     corev1.DNSClusterFirst,
+							SchedulerName:                 "default-scheduler",
+							ServiceAccountName:            "k8ssandra-operator",
```

Comment:
[question] Is it safe to hard-code this? I think so, but I'm not 100% sure that it can't be modified by kustomizations.

Reply:
I asked @adejanovski about this, and it seemed like this is fine. It looks like in other places (reaper and stargate  for example) we have access to the service account name, but I don't think we do in the MedusaClusterTemplate here.